src/subsystems/odometry/odometry_tank.cpp

```diff
@@ -50,125 +54,126 @@ OdometryTank::OdometryTank(vex::encoder &left_vex_enc, vex::encoder &right_vex_e
+        speed = ema.get_value();
+        // Calculate robot acceleration
+        accel = (speed - last_speed) / tmr.time(sec);
+
+        // Calculate robot angular velocity (deg/sec)
+        ang_speed_deg = smallest_angle(current_pos.rotation().degrees(), last_pos.rotation().degrees()) / tmr.time(sec);
+
+        // Calculate robot angular acceleration (deg/sec^2)
+        ang_accel_deg = (ang_speed_deg - last_ang_speed) / tmr.time(sec);
+
+        tmr.reset();
+        last_pos = current_pos;
+        last_speed = speed;
+        last_ang_speed = ang_speed_deg;
+    }
+
+    return current_pos;
 }
 
 /**
  * Using information about the robot's mechanical structure and sensors, calculate a new position
  * of the robot, relative to when this method was previously ran.
  */
-Pose2d OdometryTank::calculate_new_pos(robot_specs_t &config, Pose2d &curr_pos, double lside_revs, double rside_revs,
-                                       double angle_deg) {
-  Pose2d new_pos(0, 0, 0);
+Pose2d OdometryTank::calculate_new_pos(
+  robot_specs_t &config, Pose2d &curr_pos, double lside_revs, double rside_revs, double angle_deg
+) {
+    Pose2d new_pos(0, 0, 0);
 
-  static double stored_lside_revs = lside_revs;
-  static double stored_rside_revs = rside_revs;
+    static double stored_lside_revs = lside_revs;
+    static double stored_rside_revs = rside_revs;
 
-  // Convert the revolutions into "change in distance", and average the values for a "distance driven"
-  double lside_diff = (lside_revs - stored_lside_revs) * PI * config.odom_wheel_diam;
-  double rside_diff = (rside_revs - stored_rside_revs) * PI * config.odom_wheel_diam;
-  double dist_driven = (lside_diff + rside_diff) / 2.0;
+    // Convert the revolutions into "change in distance", and average the values for a "distance driven"
+    double lside_diff = (lside_revs - stored_lside_revs) * PI * config.odom_wheel_diam;
+    double rside_diff = (rside_revs - stored_rside_revs) * PI * config.odom_wheel_diam;
+    double dist_driven = (lside_diff + rside_diff) / 2.0;
 
-  double angle = angle_deg * PI / 180.0; // Degrees to radians
+    double angle = angle_deg * PI / 180.0; // Degrees to radians
 
-  // Create a vector from the change in distance in the current direction of the robot
-  // deg2rad((smallest_angle(curr_pos.rot, angle_deg)/2 + curr_pos.rot, dist_driven)
-  Vector2D chg_vec(angle, dist_driven);
+    // Create a vector from the change in distance in the current direction of the robot
+    // deg2rad((smallest_angle(curr_pos.rot, angle_deg)/2 + curr_pos.rot, dist_driven)
+    Translation2d chg_vec(angle, dist_driven);
 
-  // Create a vector from the current position in reference to X,Y=0,0
-  Translation2d curr_point(curr_pos.x(), curr_pos.y());
-  Vector2D curr_vec(curr_point);
+    // Create a vector from the current position in reference to X,Y=0,0
+    Translation2d curr_point(curr_pos.x(), curr_pos.y());
+    Translation2d curr_vec(curr_point.x(), curr_point.y());
```

Comment:
you can get rid of one of these two lines. they have the same value and kind of just clutter things up to have both